scripts/ros-generate-cache.sh

```diff
@@ -0,0 +1,132 @@
+# /bin/sh (deliberately no !#)
+#
+# Usage: cd meta-ros
+#        sh scripts/ros-generate-cache.sh ROS_DISTRO YYYYMMDD PATH-TO-LOCAL-ROSDISTRO ROSDISTRO-COMMIT
+#            or
+#        sh scripts/ros-generate-cache.sh --version
+#
+#        Generate cache.yaml file for the YYYYMMDD release of ROS_DISTRO from the specified commit of a local
+#        ros/rosdistro.git . If running prior to first release of ROS_DISTRO, specify "none" for YYYYMMDD. NB. The release
+#        date might not match the commit timestamp of ROSDISTRO-COMMIT; however, the commit timestamp is used for the
+#        DATETIME when forming the name of the created branch.
+#
+#        It will also copy the files from PATH-TO-LOCAL-ROSDISTRO/rosdep to the meta-ros and asks you to
+#        adjust /etc/ros/rosdep/sources.list.d/20-default.list before running rosdep update if it doesn't
+#        point to these files already.
+#
+# Copyright (c) 2019 LG Electronics, Inc.
+
+SCRIPT_NAME="ros-generate-cache"
+SCRIPT_VERSION="1.0.0"
+
+usage() {
+    echo "Usage: cd meta-ros"
+    echo "Usage: sh scripts/$SCRIPT_NAME.sh ROS_DISTRO YYYYMMDD PATH-TO-LOCAL-ROSDISTRO ROSDISTRO-COMMIT"
+    echo "               or"
+    echo "       sh scripts/$SCRIPT_NAME.sh --version"
+    exit 1
+}
+
+if [ $1 = "--version" ]; then
+    echo "$SCRIPT_NAME $SCRIPT_VERSION"
+    exit
+fi
+
+[ $# -ne 4 ] && usage
+
+ROS_DISTRO=$1
+ROS_DISTRO_RELEASE_DATE=$2
+ROS_DISTRO_CHECKOUT=$3
+ROS_DISTRO_COMMIT=$4
```

Comment:
The two variable above are not related to the ROS distro for which the cache is being generated => I'm going to rename them to ROS_ROSDISTRO_{CHECKOUT_PATH,COMMIT}.

Reply:
OK